### DISCUSSION TITLE
Matataki QUEST 大家谈

### DISCUSSION THREAD

**Discussion by Linklinkguan** [Matataki-io/matataki-frontend#1001] [Discussion] [answered]
[功能介绍](https://www.yuque.com/matataki/matataki/pmpeoi)
Matataki QUEST 这个DAPP已经上线了2周时间，目前一共发布了27个任务，其中3个任务的奖励已经被领取完毕了。这是我们的Alpha版本，还存在非常多不完善的地方，主要集中在任务发布之后的维护功能

- [ ] 自定义任务被完成之后没有通知
- [ ] 未被完成的任务无法被撤销&退款
- [ ] ~~发布后重编辑自定义任务详情~~
- [ ] 补充任务说明 或 任务描述版本区分
- [ ] 预览任务详情（Markdown）
- [ ] 描述部分对Markdown语法的展示上还不够显眼，比如超链接
- [ ] 集中管理我创建的任务

欢迎来这儿让我们知道你们的想法和建议，以及还希望添加哪些任务类型~

**Comment by whyouare111** [comment]
这个任务功能的想象空间挺多的，任务发布者带accessToken可以调用API来获取申请列表，同意/拒绝，这样可以结合bot做出各种任务。bot也可以开源，这样愿意提供bot服务的都可以自建，或者开放给其他人使用了。

**Reply by whyouare111** [reply]
比如如果要基于自定义任务做一个“请到Mastodon上关注我”的流程，可以这样做：
1. 请选择一个Mastodon实例注册账号
2. 请用这个账号关注我的账号
3. 请把Mastodon时间线绑定到Matataki Auth，或者在【个人信息】里维护好指向Mastodon的个人首页的链接，且在Mastodon上也维护好指向Matataki的链接

bot 需要配置上比较长期的matataki和mastodon的accessToken，或者定期用clientId和clientSecret拿新accessToken，流程则是这样的：
1. 定期获取申请列表，对每一个未处理的任务获取其申请人，再通过申请人找到映射的Mastodon账号，形成一个“申请ID-Mastodon账号键值对列表”
2. 获取自己在Mastodon的关注列表，和上面那个账号列表求交集，求出的交集就是正确完成任务的键值对，对这些ID同意；1列表-交集列表就是失败的，对这些键值对拒绝


---
这里暂时没考虑锁账号关注请求的情况。如果有开就不确认Mastodon的关注人，而是获取关注请求

**Comment by lychees** [comment, marked as answer]
感觉输入密码领 bounty 也可以做成一个「标准任务形式」。。
这样我可以去设计基于 Quest 的解谜游戏了。。。

**Reply by Linklinkguan** [reply]
啪的一下，很快啊！功能就开发出来了。新的口令红包功能已经上线
感谢 @xiaotiandada 的高效开发工作！
立即领取奖励：https://quest.matataki.io/29（仅限5席）

让我们来看下功能截图
![image](https://user-images.githubusercontent.com/10827774/104027155-f42d8180-5201-11eb-8d7e-86a3927630bb.png)
![image](https://user-images.githubusercontent.com/10827774/104027295-263ee380-5202-11eb-8e52-4c33a55054ad.png)
![image](https://user-images.githubusercontent.com/10827774/104027232-0c050580-5202-11eb-8100-036a95dcedf3.png)

**Reply by Linklinkguan** [reply]
这个模式下存在极大的作弊（薅羊毛）可能性，当下似乎没那么重要，不过长期来看这的确是个问题

**Reply by whyouare111** [reply]
看怎么用。普通的猜谜估计顶不过搜索引擎，形成专门的群固定答案的也肯定是一下就被搞光吧。用作Flag管理的密码倒是可以每个人看到的东西是不一样的，加盐签名的结果作为口令之类。如果这个动作在服务端做也不会泄漏

**Comment by lychees** [comment]
wow

**Comment by whyouare111** [comment]
如果某些题目的谜面要用其他题得到的token解锁查看，可以构造一个任务链了。
比如一种没有其他获取途径，只能答题获得的fan票，第一题就是这样新年快乐的🧧性质，拿到就可以解锁其他任务了。
持有1个可以答Lv1的题，通过答Lv1的题获得5个可以答Lv2的题，Lv2的回报更多。
集到足够的token可以开NFT宝箱，NFT可以是Dego那种铲子性质的，引导到其他类型的/合作的DAPP等等。
铲子挖出的token可以用于解锁某些资源作为奖励，也可以在Quest里体现

**Comment by Linklinkguan** [comment]
![image](https://user-images.githubusercontent.com/10827774/104131669-d5bab800-53b2-11eb-80f3-b0ad36c49189.png)
关于奖励领取的描述看起来比较令人费解，“总奖励数量”搞不清楚代表的是Fan票数量，还是领取的份数。同时“总奖励数量”的分数左边数值和右边数值并不能很容易被感知，哪个是总量哪个是余量。

**Reply by whyouare111** [reply]
还可领取的份数/总份数

**Reply by Linklinkguan** [reply]
已经将“总奖励数量”调整为了“奖励份数”

**Comment by Linklinkguan** [comment]
如果能够将QUEST融入到在Matataki或是社交媒体的发布内容流程中会怎样呢？
比如：当我成功在Matataki发布一篇文章之后，页面中提示我是否需要通过QUEST来吸引更多人来阅读，于是我就可以直接在文章页面中完成QEUST的发布，作为一种运营手段。如果将其拓展到Matataki的一个插件功能生态中，则可以实现在发布Twitter的时候做到相同的事情。
